src/pages/material/create-color/create-color.js

```diff
@@ -0,0 +1,273 @@
+      .required(materialErrorMessages.VALIDATION_ERROR),
+
+    enSimpleName: Yup.string()
+      .min(2, materialErrorMessages.MIN_LENGTH_MESSAGE)
+      .max(100, materialErrorMessages.MAX_LENGTH_MESSAGE)
+      .required(materialErrorMessages.VALIDATION_ERROR),
+
+    code: Yup.string()
+      .min(1, colorErrorMessages.MIN_LENGTH_MESSAGE)
+      .max(8, colorErrorMessages.MAX_CODE_LENGTH_MESSAGE)
+      .matches(
+        config.formRegExp.colorCode,
+        colorErrorMessages.CODE_VALIDATION_ERROR
+      )
+      .required(colorErrorMessages.VALIDATION_ERROR)
+  });
+
+  const {
+    values,
+    handleChange,
+    handleSubmit,
+    errors,
+    touched,
+    setFieldValue,
+    setFieldError
+  } = useFormik({
+    validationSchema: formSchema,
+    validateOnBlur: true,
+    initialValues: {
+      ...formikValues,
+      code: '',
+      colorImage: '',
+      image: '',
+      available: false
+    },
+    onSubmit: (data) => {
+      const { colorImage, image, ...rest } = data;
+      const color = createColor(rest);
+
+      const foundCode = colors.map((item) => item.code);
+      if (!colorImage || !image) {
+        return;
+      }
+      if (foundCode.includes(+rest.code)) {
+        setFieldError('code', config.colorErrorMessages.CODE_NOT_UNIQUE_ERROR);
+        return;
+      }
+      dispatch(setNewColorToStore(color));
+      addNewColorImages(colorImage);
+      dispatch(showColorDialogWindow(false));
+      setImagesToUpload(image);
+    }
+  });
+
+  const tabPanels = languages.map((lang, index) => (
+    <TabPanel key={lang} value={tabsValue} index={index}>
+      <Paper className={styles.materialItemAdd}>
+        <TextField
+          data-cy={`${lang}Name`}
+          id={`${lang}Name`}
+          className={styles.textfield}
+          variant='outlined'
+          label={config.labels.colors.name}
+          error={touched[`${lang}Name`] && !!errors[`${lang}Name`]}
+          multiline
+          value={values[`${lang}Name`]}
+          onChange={handleChange}
+        />
+        {touched[`${lang}Name`] && errors[`${lang}Name`] && (
+          <div className={styles.inputError}>{errors[`${lang}Name`]}</div>
+        )}
+        <TextField
+          data-cy={`${lang}SimpleName`}
+          id={`${lang}SimpleName`}
+          className={styles.textfield}
+          variant='outlined'
+          label={config.labels.colors.simpleName}
+          multiline
+          error={touched[`${lang}SimpleName`] && !!errors[`${lang}SimpleName`]}
+          value={values[`${lang}SimpleName`]}
+          onChange={handleChange}
+        />
+        {touched[`${lang}SimpleName`] && errors[`${lang}SimpleName`] && (
+          <div className={styles.inputError}>{errors[`${lang}SimpleName`]}</div>
+        )}
+      </Paper>
+    </TabPanel>
+  ));
+
+  if (loading) {
+    return <LoadingBar />;
+  }
+
+  const languageTabs =
+    languages.length > 0
+      ? languages.map((lang, index) => <Tab label={lang} key={index} />)
```

Comment:
Don't use index for key property, use something stable like **id** or **name**

Reply:
fixed